Test/traits/.gitignore

```diff
@@ -0,0 +1 @@
+TraitCompile/
```

Comment:
It would be good to rename these Java directories (to, say, `TraitCompile-java`) so they more easily can be excluded. But I'm not suggesting putting that in this same PR.